tests/drivers/can/api/src/canfd.c

```diff
@@ -425,6 +544,47 @@ ZTEST_USER(canfd, test_invalid_sample_point)
 	zassert_equal(err, -EINVAL, "invalid sample point of 100.0% accepted (err %d)", err);
 }
 
+/**
+ * @brief Test using an invalid bitrate.
+ */
+ZTEST_USER(canfd, test_invalid_bitrate)
+{
+	struct can_timing timing;
+	int err;
+
+	err = can_calc_timing_data(can_dev, &timing, 0, TEST_SAMPLE_POINT);
+	zassert_equal(err, -EINVAL, "invalid bitrate of 0 bit/s accepted (err %d)", err);
+
+	err = can_calc_timing_data(can_dev, &timing, 8000001, TEST_SAMPLE_POINT);
+	zassert_equal(err, -EINVAL, "invalid bitrate of 8000001 bit/s accepted (err %d)", err);
+}
+
+/**
+ * @brief Test can_calc_timing_data() with very low/high sample point value.
+ */
+ZTEST_USER(canfd, test_sample_point_low_high)
+{
+	struct can_timing timing;
+	int err;
+
+	const struct can_timing * min_timing = can_get_timing_data_min(can_dev);
+	const struct can_timing * max_timing = can_get_timing_data_max(can_dev);
+
+	err = can_calc_timing_data(can_dev, &timing, TEST_BITRATE_1, TEST_SAMPLE_POINT_LOW);
+	zassert_ok(err, "failed to calculate timing (err %d)", err);
+	zassert_true(timing.phase_seg1 >= min_timing->phase_seg1,
+		"returned phase_seg1 is too low");
+	zassert_true(timing.phase_seg2 <= max_timing->phase_seg2,
+		"returned phase_seg2 is too high");
+
+	err = can_calc_timing_data(can_dev, &timing, TEST_BITRATE_1, TEST_SAMPLE_POINT_HIGH);
+	zassert_ok(err, "failed to calculate timing (err %d)", err);
+	zassert_true(timing.phase_seg1 <= max_timing->phase_seg1,
+		"returned phase_seg1 is too high");
+	zassert_true(timing.phase_seg2 >= min_timing->phase_seg2,
+		"returned phase_seg2 is too low");
+}
```

Comment:
This is tested in the `tests/drivers/can/timing/` test suite.

Reply:
My mistake, I was calculating code coverage only from api test suite.
Removed.